internal/app/ui/cpwsarea/wsmap/pmap/psettings/screenshot.go

```diff
@@ -124,14 +128,31 @@ func (p *Panel) ProcessUnit(u unit.Unit) bool {
 	return p.app.PathsFilter().IsVisiblePath(u.Instance().Prefab().Path())
 }
 
-func saveScreenshot(pixels []byte, w, h int) error {
+func (p *Panel) saveScreenshot(pixels []byte, w, h int) error {
 	if err := os.MkdirAll(cfg.ScreenshotDir, os.ModeDir); err != nil {
 		log.Print("unable to create screenshot directory:", err)
 		return err
 	}
 
 	out, _ := os.Create(cfg.ScreenshotDir + "/" + time.Now().Format(util.TimeFormat) + ".png")
+	if p.sessionScreenshot.toClipboardMode {
+		defer fileToClipboardAndDelete(out.Name())
```

Comment:
This relies on the assumption that the user has set a screenshot directory, which is somewhat counterintuitive. As a user, I don't expect that a screenshot directory is required when saving to the clipboard. 

I suggest changing the flow as follows:

- When the user selects the "To Clipboard" option, the input for the screenshot directory becomes disabled.
- During the save process, we need to check whether we are saving to the directory or to the clipboard.
  - If it's the former, proceed as it was done previously.
  - If it's the latter, use a separate method that saves the screenshot to the user's temporary directory and then copies it to the clipboard.

Don't mix the saving flows; they function and behave differently, so one should not depend on the other.

Reply:
![image](https://github.com/SpaiR/StrongDMM/assets/70376633/a2ce6d73-a7c5-46f0-a432-92112e1dcb50)